qiskit/visualization/circuit/matplotlib.py

```diff
@@ -759,11 +946,48 @@ def _draw_regs_wires(self, num_folds, xmax, max_x_index, qubits_dict, clbits_dic
                     zorder=PORDER_TEXT,
                 )
 
+    def _add_nodes_and_coords(
+        self, nodes, node_data, wire_map, layer_widths, qubits_dict, clbits_dict, glob_data
+    ):
+        """Add the nodes from ControlFlowOps and their coordinates to the main circuit"""
+        for flow_drawer in self._flow_drawers.values():
+            for i in range(0, len(flow_drawer)):
+                if flow_drawer[i] is None:  # No else
+                    continue
+
+                nodes += flow_drawer[i]._nodes
+                flow_drawer[i]._get_coords(
+                    node_data,
+                    wire_map,
+                    layer_widths,
+                    qubits_dict,
+                    clbits_dict,
+                    glob_data,
+                    flow_parent=flow_drawer[i]._flow_parent,
+                    is_if_while=(i == 0),
```

Comment:
What's `is_if_while` actually measuring here?  It looks like it's more like a test of `is_first_block`, and if so, is it not important which number block it is _other_ than "zero" or "not zero"?

Reply:
Right. This is used in `get_coords` to do the extra width calcs for Else and Case. At that point in the code we just need to know that we're working with a circuit that's in an Else or Case and not If, While, For, or Switch. So `is_first_block` is a better name or `is_not_first_block` to go the other way.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/10207/commits/8d232e92c08e286f6ee30c1621243446e7f126e2.